[PyPer][BE] Fix test_scripted_module in StatCollector

PaulZhang12 opened the pull request:
Summary: D41985889 removed the cast to int for the inputs to torch.histc below, allowing the inputs to still be tensors. These tensors still have require_grad_ set to True, causing issues with the call to torch.histc.

Test Plan: buck2 test 'fbcode//mode/opt' fbcode//dper3/dper3/modules/low_level_modules/tests:stat_collector_test -- --exact 'dper3/dper3/modules/low_level_modules/tests:stat_collector_test - test_scripted_module (dper3.dper3.modules.low_level_modules.tests.stat_collector_test.StatCollectorTest_1)'

Differential Revision: D48800879

PaulZhang12 commented:
/easycla

PaulZhang12 commented on torch/ao/quantization/observer.py:
```diff
@@ -1175,6 +1175,10 @@ def forward(self, x_orig: torch.Tensor) -> torch.Tensor:
             ), "histogram min/max values must be scalar."
+
+            # set requires_grad to False for these tensors for use in torch.histc
+            combined_min, combined_max = combined_min.detach(), combined_max.detach()
```
This is not related to x or x_orig. This is for combined_min and combined_max tensors to function below L1182

jerryzh168 replied:
oh, maybe we should detach `self.min_val` and `self.max_val` in L1146-1147?

PaulZhang12 replied:
Yeah, we could do that. We only run into this problem in this branch though, so figured just keep it here

jerryzh168 replied:
L1199 actually did this detach, so it might make sense to move this earlier I think

PaulZhang12 replied:
detaching min_val/max_val did not work, as new tensors get introduced

PaulZhang12 committed:
[PyPer][BE] Fix test_scripted_module in StatCollector (#108232)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/108232

D41985889 removed the cast to int for the inputs to torch.histc below, allowing the inputs to still be tensors. These tensors still have require_grad_ set to True, causing issues with the call to torch.histc.

Reviewed By: jerryzh168

Differential Revision: D48800879

fbshipit-source-id: e36bbd9c63292308be9c8884aebe0f71448ffedc
